src/core/ext/client_channel/client_channel.c

```diff
@@ -977,20 +1063,47 @@ static void start_transport_stream_op_locked_inner(grpc_exec_ctx *exec_ctx,
   add_waiting_locked(calld, op);
 }
 
-static void cc_start_transport_stream_op_locked(grpc_exec_ctx *exec_ctx,
-                                                void *arg,
-                                                grpc_error *error_ignored) {
-  GPR_TIMER_BEGIN("cc_start_transport_stream_op_locked", 0);
+static void on_complete_locked(grpc_exec_ctx *exec_ctx, void *arg,
+                               grpc_error *error) {
+  grpc_call_element *elem = arg;
+  channel_data *chand = elem->channel_data;
+  call_data *calld = elem->call_data;
+  if (chand->retry_throttle_data != NULL) {
+    if (error == GRPC_ERROR_NONE) {
+      grpc_server_retry_throttle_data_record_success(
+          &chand->retry_throttle_data);
+    } else {
+      // TODO(roth): In a subsequent PR, check the return value here and
+      // decide whether or not to retry.
+      grpc_server_retry_throttle_data_record_failure(
```

Comment:
This may be already encompassed by the above TODO, but may be a good idea to also note here that only failures with a retryable or non-fatal status code and RPCs that received "do not retry" pushback should be counted for throttling purposes.

(last paragraph of https://github.com/ncteisen/proposal/blob/75e08fa10405430e8177cfd91bf63a25ff4ad617/A6.md#throttling-retry-attempts-and-hedged-rpcs)